### DISCUSSION TITLE
Use plugin without XCode

### DISCUSSION THREAD

**Discussion by zurcherart** [Vinzent03/obsidian-git#358] [Q&A]
Similar to #251, I would like to run this plugin under MacOS with out installing XCode.

I use git on this machine all the time having installed git command line tools already from another source.  

I can run git from the command line and get normal results.  I can also run applications like GitHub Desktop and SourceTree which use the git comman-line tools as far as I know.

My git command line tools are installed in /usr/local/bin/git

my git --version is 2.15.0

Is there a way to use the github community plugin with Obsidian with git command line tools that weren't installed by Xcode?

Cheers!

**Comment by Vinzent03** [comment]
Should be working just fine. I guess it can't find git. Can you try specifying the git binary path manually in the settings?

**Comment by engage-results** [comment]
Hey!  Thanks for the quick answer. You are right, I needed to specify the binary path but I had to update my old release of obsidian to get to the settings options ... I didn't realise I was so far behind.  Everything looks good, except I can't login to github.

![Screenshot 2022-10-31 at 13 12 05](https://user-images.githubusercontent.com/46592966/199013174-f7ab3744-a6b3-4bf1-8bb3-47cf5d12ea82.png)

I think I must set up something for my ssh key ... but I'm a little lost with the different options. Any help is appreciated.

**Reply by Vinzent03** [reply]
Seems like it can't find your ssh management? binary as well. I need to add a setting to add custom PATH paths.

**Reply by engage-results** [reply]
Makes sense. 

How does your an application determine the path of executables by default?

My terminal shell path defaults to:

PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin

And this is determined by the file in /etc/paths and the files in /etc/paths.d. I installed a TeX system and it apparently created a file called TeX in /etc/paths.d which contains the path to the TeX commands. (That's unrelated to Obsidian ... but tracking it down helped me figure out how my shell's paths were set. 

I can't figure out if apps like Obsidian can get the system PATH or if it's only used for shells.

I guess I'm thinking if there is a way to make sure the default system PATH is seen by obsidian app so that when I use the plugin it finds the right commands I could do that. 

Now we've reached the end of my knowledge on this topic :D (we didn't have far to go).

It would be very helpful if you're able to put this into plugin settings.  But, in the meantime I'm happily using git from the command line and github-desktop. Will be extra sauce when I can do it in the obsidian plugin.

Thanks!

**Reply by Vinzent03** [reply]
I think it depends on the way Obsidian is installed and the os. Adding a setting will solve that.

**Reply by Vinzent03** [reply]
I added a setting to add paths to the PATH env var. May you try it?

**Comment by engage-results** [comment]
HI Vinzent ... that worked thanks very much!

There was another issue.  

![Screenshot 2022-11-02 at 16 09 57](https://user-images.githubusercontent.com/46592966/199534041-6a3cfb27-cdb2-4198-b2f3-bb41efaaaee7.png)

I kept getting this message. It's not your plugin's "fault", but it only showed up in your plugin so letting you know about it because other people might have the same error if they haven't already corrected it on their system.

I had forgotten how I'd installed git on this machine. But I retraced my steps.

My git install came from the notes at the official source at https://git-scm.com/download/mac and then ultimately I used Tim Harper's installer from here https://sourceforge.net/projects/git-osx-installer/

Digging around a bit on StackOverflow I learned that some lines in the standard gitconfig file that's installed at /usr/local/git/etc specify a path using the tilde (ie. ~/.gitcinclude which is the file being referenced in the error message). I believe this would work under Mavricks but in Catalina (and above?) this doesn't work for git that is called by a "subprocess". On StackOverflow the questions was related to Apache running git in a subprocess. 

So the configuration that shipped with the git installer works when git is run from the command line, but it fails with the error message I was seeing when it's run as a subprocess.  Presumably, the Obsidian-Git plugin runs git as a subprocess.

The fix was to edit /usr/local/git/etc/gitconfig and change the paths to use $HOME instead of ~.

For example, the second line should be 

 excludesfile = $HOME/.gitignore 
  not
 excludesfile = ~/.gitignore

I don't think that's something you can handle in your plugin. But if other people might run into the same issue.

Now everything seems to work!

Thanks for helping out so quickly.

Cheers,
Steve

**Reply by Vinzent03** [reply]
That's great. I had to change something. Does it still work in 2.9.1?

**Comment by engage-results** [comment]
Yes ... it still works! Thanks.
